### PR TITLE
USHIFT-1349: automate publishing microshift pre-release info

### PR DESCRIPTION
**Pull request by dhellmann**

### CODE REVIEW

**Comment by dhellmann** [comment]
/pj-rehearse periodic-ci-openshift-microshift-main-publish-release-notes

**Comment by dhellmann** [comment]
@pmtk @pacevedom I believe this is ready for your review. We could wait until there's a new EC and use the rehearsal job to test it, if you want, or we could just merge it and see if we get an error.

File: ci-operator/step-registry/openshift/microshift/publish-release-notes/openshift-microshift-publish-release-notes-commands.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -xeuo pipefail
+
+if [[ "$JOB_NAME" == rehearse* ]]; then
+    echo "INFO: \$JOB_NAME starts with rehearse - running in DRY RUN mode"
+    export DRY_RUN=y
+fi
```
Comment (pmtk) [inline ci-operator/step-registry/openshift/microshift/publish-release-notes/openshift-microshift-publish-release-notes-commands.sh:7]: I don't see `gen_ec_release_notes.py` making use of `DRY_RUN` env var

Comment (dhellmann) [reply ci-operator/step-registry/openshift/microshift/publish-release-notes/openshift-microshift-publish-release-notes-commands.sh]: I've updated the implementation here to pass the command line flag

**Pull request merged**